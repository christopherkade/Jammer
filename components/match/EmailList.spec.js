import { shallowMount } from '@vue/test-utils'
import cmp from './EmailList.vue'

describe('[UNIT] Testing the /match Email List component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(cmp, {
      computed: {
        userList() {
          return [
            {
              email: 'mockedEmail1'
            }
          ]
        },
        userEmail() {
          return 'mockedEmail'
        }
      }
    })
  })

  it('Snapshot testing', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Component validation', () => {
    it('should render the main (parent) component', () => {
      expect(wrapper.is(cmp)).toBe(true)
    })
  })

  describe('Content validation', () => {
    it('should render the user name tag', () => {
      expect(wrapper.text()).toContain('mockedEmail1')
    })
  })
})
