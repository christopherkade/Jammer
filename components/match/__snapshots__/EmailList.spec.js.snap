// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UNIT] Testing the /match Email List component Snapshot testing 1`] = `
"<div class=\\"columns\\"><div><div class=\\"column\\"><span class=\\"email-tag\\">
        mockedEmail1
        <button aria-label=\\"delete\\" class=\\"delete\\"></button></span></div></div></div>"
`;
